client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -75,10 +75,14 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 	const currentStepRoute = params.step || '';
 	const isLoggedIn = useSelector( isUserLoggedIn );
 	const { lang = null } = useParams();
+	const isValidStep = params.step != null && stepPaths.includes( params.step );
 
 	// Start tracking performance for this step.
 	useStartStepperPerformanceTracking( params.flow || '', currentStepRoute );
-	useFlowAnalytics( { flow: params.flow, step: currentStepRoute, variant: flow.variantSlug } );
+	useFlowAnalytics(
+		{ flow: params.flow, step: params.step, variant: flow.variantSlug },
+		{ enabled: isValidStep }
```

Comment:
Something isn't connecting with me here. To clarify my undestanding: A step being `invalid` here covers the case where the request is for `/setup/:flow/:lang`, correct? Where the step is falsely assumed to be `lang`. In that case, we "disable" a hook.

If my reading is correct, then the issue still exists in the app (in the routing/`useFlowNavigation`), so are we ignoring the root cause? Should we focus instead on attaching a correct/valid step onto `params`, so this ambiguity is eliminated (and the need to accompany with `isValidStep`)?

Reply:
> I think the ideal solution would be not to render the complete flow-render code. We may have bugs in any code inside this component regardless of whether the `useFlowNavigation` returns the correct step because the flows always assume a valid step was rendered.
>
> My concern in updating `useFlowNavigation`, which are flows we are checking if the user is on the initial step, like [this](https://github.com/Automattic/wp-calypso/blob/ce12270095c0bef77d1986e8115a366bf16f1956/client/landing/stepper/declarative-flow/migration/index.tsx#L320).

Agreed. I meant to explain my thinking in broader terms. I haven't actually inspected `useFlowNavigation`, just thought this is an issue with a layer above the `FlowRender` component e.g. a "routing" issue or a missing knowledge of a default or first step. 

I guess a question is whether a flow access without a valid step makes sense at all in purely logical terms. There's obviously always some step being rendered - so what's a flow with an invalid one? If this is correct, then what you suggest to "skip the `FlowRender` component" makes sense. Is this your thinking also?

If you'd rather we ship this and do a follow-up in separate PR/issue, also works for me.

Reply:
>  Is this your thinking also?

Yes, I think it will save us to introduce more complexity and prevent bugs.

>  If you'd instead we ship this and do a follow-up in separate PR/issue, also works for me. 
Agree. I am working on another project now, so shipping this will give me more time to do it. 

I just need approval on this PR.